client/client.go

```diff
@@ -514,25 +510,22 @@ func parseErrorResponse(errResponse *http.Response) error {
 	code := errResponse.StatusCode
 	text := http.StatusText(code)
 
+	errString := fmt.Errorf("%s. %d %s", errResponseNotOK, code, text)
+
 	body, err := io.ReadAll(errResponse.Body)
 	if err != nil {
-		return err
+		return errString
 	}
 
 	var errorMap map[string]string
 	err = json.Unmarshal(body, &errorMap)
 	if err != nil {
-		return err
+		return errString
```

Comment:
The actual error from io.ReadAll and json.Unmarshal are being lost with this change returning errString.

Reply:
Yes that's intended.

Reply:
If either of those errors out it means there was an issue with the response body itself (which  contains a more detailed error message in production cases). In that case we still want to get at the very least an error message containing a the status code and status text.

Reply:
That said, this won't actually happen in production as in actual usage of PHD the errors are always sent using the `utils-go` client that adds a detailed error message to the response body.

This PR is mainly just to fix an issue that was happening when using the client in unit tests where the response body is missing the body.